### ISSUE TITLE
Cross-entropy broken for small training set or batches?

### ISSUE THREAD

**Issue by emjotde** [rsennrich/nematus#14]
Hi,
when using the WMT de-en model to initialize a training run on a single sentence corpus, I get very incorrect cross-entropy results. I am using only default setting, no dropout or anything. I verified that the model is being loaded correctly. The values in the final layer before it goes into cross-entropy are also correct.

For instance for the pair

das ist ein Test .
this is a test .

The cost should be around 0.61868, verified by scoring with Amun. Nematus in its current version from master produces 102 for the first forward step. When repeating this sentence 20 times and increasing the batch size accordingly, the cost becomes 111.15, which makes no sense as it should at least average to the same cost as for the single sentences. 

After inspecting the cost vector manually, it seems only the first value is being calculated correctly. Can anyone confirm this? Or is something wrong with my setup?

**Comment by emjotde** [comment]
The same happens with bleeding edge Theano.

**Comment by emjotde** [comment]
OK, solved. This was my own mistakes. Nematus does not check wether manually set `n_words` corresponds to the dimensions of the output embeddings. However it uses `n_words` to select target ids after softmax. The result is that probabilities after the first word a read at the wrong position. 

Feature request: Check if `n_words` and Wemb_dec,shape[1] correspond.
